pkg/hardware/config.go

```diff
@@ -0,0 +1,98 @@
+package hardware
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	pbnjv1alpha1 "github.com/tinkerbell/cluster-api-provider-tinkerbell/pbnj/api/v1alpha1"
+	tinkv1alpha1 "github.com/tinkerbell/cluster-api-provider-tinkerbell/tink/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/yaml"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+)
+
+type HardwareConfig struct {
+	hardwareList []tinkv1alpha1.Hardware
+	bmcList      []pbnjv1alpha1.BMC
+	secretList   []corev1.Secret
+}
+
+func (hc *HardwareConfig) GetHardwareConfig(hardwareFileName string) error {
+	err := hc.parseHardwareConfig(hardwareFileName)
+	if err != nil {
+		return fmt.Errorf("unable to parse hardware file %s: %v", hardwareFileName, err)
+	}
+	return nil
```

Comment:
should we return the hardwareConfig struct here?

Reply:
Actually since its a method of a struct, the changes are kinda made by pass by reference.
So `hc` would have the parsed hardware config.

Reply:
Yea sure 👍